components/search/search.tsx

```diff
@@ -253,7 +253,6 @@ const Search: React.FC<Props> = (props: Props): JSX.Element => {
         }
 
         if (props.isMentionSearch) {
-            e.preventDefault();
```

Comment:
Was preventing the default form submission. When `enter` is hit the second `props.isMentionSearch` is false and submission works.